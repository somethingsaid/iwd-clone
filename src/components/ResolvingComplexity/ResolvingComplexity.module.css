.ImageBackground {
  min-height: 300px;
  background: url(https://images.unsplash.com/photo-1555949963-ff9fe0c870eb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2850&q=80) no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  display: flex;
}

.ImageBackground div {
  margin: 65px 50px;
}

.description {
  display: flex;
  flex-direction: column;
  align-content: center;
  flex: 1;
  padding: 0 25px;
  background-color: white;
}

.description h1 {
  font-size: 5.0vw;
  font-weight: 400;
}

.description h3 {
  font-size: 2.5vw;
  font-weight: 400;
}

.description a {
  text-decoration: none;
  color: #007cba;
}

.placeholder {
  display: none;
}

.imageCredit {
  color: white;
  font-size: x-small;
}

.imageCredit a {
  text-decoration: none;
  color: #007cba;
}

@media (min-width: 768px) {
  .placeholder {
    display: block;
    flex: 2;
  }
}