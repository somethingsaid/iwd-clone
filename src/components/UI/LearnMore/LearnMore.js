import React from "react";

const learnMore = (props) => {
  return (
    <a
      href={props.url}
      target="_blank"
      rel="noopener noreferrer"
      style={{ color: "#007cba", textDecoration: "none" }}
    >
      Learn More
    </a>
  );
};

export default learnMore;
