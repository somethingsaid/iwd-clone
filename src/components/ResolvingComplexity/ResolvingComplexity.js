import React from "react";
import LearnMore from "../UI/LearnMore/LearnMore";
import classes from "./ResolvingComplexity.module.css";

const resolvingComplexity = () => {
  return (
    <div className={classes.ImageBackground}>
      <div className={classes.description}>
        <h1>Resolving complexity is our specialty</h1>
        <h3>
          The quality of Intelliwareâ€™s work delivers meaningful business value
          and enduring competitive advantage for the clients we serve. Join our
          roster of clients and modernize the way you do business.
        </h3>
        <LearnMore url="https://www.intelliware.com/focus-areas/" />
      </div>
      <div className={classes.placeholder}></div>
    </div>
  );
};

export default resolvingComplexity;
