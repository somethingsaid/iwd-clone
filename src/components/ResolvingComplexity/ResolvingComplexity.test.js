import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

import ResolvingComplexity from "./ResolvingComplexity";
import LearnMore from "../UI/LearnMore/LearnMore";

configure({ adapter: new Adapter() });

describe("<ResolvingComplexity/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ResolvingComplexity />);
  });

  it("should render a <h1>, <h3>, and a <LearnMore />", () => {
    expect(wrapper.find("h1")).toHaveLength(1);
    expect(wrapper.find("h3")).toHaveLength(1);
    expect(wrapper.find(LearnMore)).toHaveLength(1);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ResolvingComplexity />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
