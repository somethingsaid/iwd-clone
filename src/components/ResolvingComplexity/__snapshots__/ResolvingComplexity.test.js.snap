// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResolvingComplexity/> renders correctly 1`] = `
<div
  className="ImageBackground"
  title="Photo by Shahadat Rahman via Unsplash"
>
  <div
    className="description"
  >
    <h1>
      Resolving complexity is our specialty
    </h1>
    <h3>
      The quality of Intelliwareâ€™s work delivers meaningful business value and enduring competitive advantage for the clients we serve. Join our roster of clients and modernize the way you do business.
    </h3>
    <a
      href="https://www.intelliware.com/focus-areas/"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#007cba",
          "textDecoration": "none",
        }
      }
      target="_blank"
    >
      Learn More
    </a>
  </div>
  <div
    className="placeholder"
  />
</div>
`;
